python/tank/util/storage_roots.py

```diff
@@ -366,22 +368,38 @@ def get_local_storages(self, sg_connection):
 
         # create the SG connection and query
         log.debug("Querying SG local storages...")
-        sg_storages = sg_connection.find("LocalStorage", [], local_storage_fields)
-        log.debug("Query returned %s storages." % (len(sg_storages)))
+
+        # execute the get_storage_roots core hook
+        try:
+            from .. import sgtk_from_path
+
+            tk = sgtk_from_path(config_path.current_os)
+
+            sg_storages = tk.execute_core_hook_method(
+                constants.GET_STORAGE_ROOTS_HOOK_NAME,
+                "execute",
+                sg_connection=sg_connection,
+                local_storage_fields=local_storage_fields,
+                project_id=project_id,
+            )
+        except Exception as e:
+            # Catch errors to not kill our thread, log them for debug purpose.
+            log.debug(
+                "%s hook failed with %s" % (constants.GET_STORAGE_ROOTS_HOOK_NAME, e)
+            )
 
         # create lookups of storages by name and id for convenience. we'll check
         # against each root's shotgun_storage_id first, falling back to the
         # name if the id mapping field is not defined.
         sg_storages_by_id = {}
         sg_storages_by_name = {}
         for sg_storage in sg_storages:
-            id = sg_storage["id"]
-            name = sg_storage["code"]
+            id = sg_storage.get("id")
+            name = sg_storage.get("code")
```

Comment:
If `get` is used and the key does not exist, this will return `None`. Can this affect the following lines where `id` and `name` values are needed?